### ISSUE TITLE
Make "http" commands verbose

### ISSUE THREAD

**Issue by ineu** [nushell/nushell#9912]
### Related problem

I was trying to use "http get" instead of curl. For some reason an external API didn't accept my headers like it did with curl. I was not able to ensure that "http get" does send the specified headers because the API uses https, so I can't use tcpdump, and "http get" doesn't have a "verbose"/"trace" option to show what it is doing

### Describe the solution you'd like

Add the "verbose mode" to "http" commands, similar to `curl -v`. If specified, "http" should print sent/received headers to stdout.

**Comment by amtoine** [comment]
@ineu 
could you give a concrete example with an `http get` call, its current output and the desired output?
that would be super helpful to see what can be done :innocent:

**Comment by ineu** [comment]
Curl:
```
(
    curl -Ss
      -H 'Content-Type: application/json'
      -H $"Authorization: Bearer ($env.CLOUDFLARE_API_TOKEN)"
      $"https://api.cloudflare.com/client/v4/zones/($zone_id)/dns_records"
)
```

http get:
```
  (
    http get
      -e
      -H [Content-Type application/json]
      -H [Authorization $"Bearer ($env.CLOUDFLARE_API_TOKEN)"]
      $"https://api.cloudflare.com/client/v4/zones/($zone_id)/dns_records"
  )
```

The curl version returns json body from Cloudflare. The http version returns an error `Missing X-Auth-Key, X-Auth-Email or Authorization headers`, which makes me think that the headers are not properly passed.
The headers values are fine, I've checked them via `print [Authorization $"Bearer ($env.CLOUDFLARE_API_TOKEN)"]`

**Comment by fdncred** [comment]
I'm not following what you're asking the output to look like with a `--verbose/-v` flag. Also, I'm not sure cloudflare is a good example since it's not something people can easily test with. Is there some example site that allows testing like this that is free?

**Comment by fdncred** [comment]
I'm wondering
1. am i looking in the right place?
2. does http get just not support bearer tokens? 
https://github.com/nushell/nushell/blob/a98b3124c52262d0a72c4d009cfc86ee0977bb0d/crates/nu-command/src/network/http/client.rs#L105-L137

**Comment by fdncred** [comment]
ok, i finally understand a bit better. The `request_add_authorization_header` isn't really in play here because that's really only useful when passing `-u username -p password`. In this case, we're passing the bearer token via `request_add_custom_headers`.

I made these changes to `crates/nu-command/src/network/http/get.rs`.
I was able to add this code before the `send_request()` call.
```rust
    let header_names = request.header_names();
    for header_name in header_names {
        eprintln!("{}: {:?}", header_name, request.header(&header_name));
    }
```
and this after the `send_request()` call.
```rust
    request_handle_response_headers(span, response)
```
and get this
```nushell
❯ http get -e -H {Content-Type: application/json, Authorization: $"Bearer 12345"} https://www.example.com
authorization: Some("Bearer 12345")
content-type: Some("application/json")
╭───┬───────────────┬───────────────────────────────╮
│ # │     name      │             value             │
├───┼───────────────┼───────────────────────────────┤
│ 0 │ age           │ 42793                         │
│ 1 │ cache-control │ max-age=604800                │
│ 2 │ content-type  │ text/html; charset=UTF-8      │
│ 3 │ date          │ Fri, 04 Aug 2023 13:11:25 GMT │
│ 4 │ etag          │ "3147526947+gzip"             │
│ 5 │ expires       │ Fri, 11 Aug 2023 13:11:25 GMT │
│ 6 │ last-modified │ Thu, 17 Oct 2019 07:18:26 GMT │
│ 7 │ server        │ ECS (dab/4B67)                │
│ 8 │ vary          │ Accept-Encoding               │
│ 9 │ x-cache       │ HIT                           │
╰───┴───────────────┴───────────────────────────────╯
```
This is about as far as I can take this since I don't have a cloudflare id to test this particular situation. You can put my code snippets in the nushell codebase on your end and try it out though.

**Comment by ineu** [comment]
@fdncred Awesome, thank you!

The cloudflare part is not important at all. My initial request was not to fix the Authorization header's problem, but to make `http get` more verbose if needed. Like I wrote:

> If specified, "http" should print sent/received headers to stdout.

Though, obviously I meant stderr, as stdout should be used for the body

**Comment by NotTheDr01ds** [comment]
@ineu If I'm understanding the situation correctly (and I admittedly "skimmed" through here), you can see all of the *response* headers with `http get --full` (`-f`).  However, I agree that it would also be useful to have a way to see the *request* as it was sent.  I too ran into this type of situation when debugging authorization with an API recently.

**Comment by ineu** [comment]
That's correct: in my case I need to see the request headers. That's how it works in curl:
```
curl -v https://google.com
...
> GET / HTTP/2
> Host: google.com
> User-Agent: curl/8.2.1
> Accept: */*
>

...

< HTTP/2 301
< location: https://www.google.com/
< content-type: text/html; charset=UTF-8
...
```
So if I specified additional headers I would see that the client really sends them to the server

**Comment by fdncred** [comment]
TIL `--full` LOL. Maybe `full` should have the request headers too? Seems like that would solve this problem if anyone wants to put up a PR.

**Comment by fdncred** [comment]
Agreed. That's what I'm saying too, put them in a separate key.
